util/timer.h

```diff
@@ -156,6 +158,8 @@ class Timer {
     }
 
     while (!heap_.empty()) {
+      const FunctionInfo& top = *(heap_.top());
+      WaitForTaskCompleteIfNecessary(top);
       heap_.pop();
```

Comment:
There's a race condition during shutdown that it may `pop` a running function. As the `heap_` may change during `cond_var_.wait()` inside of `WaitForTaskCompleteIfNecessary()`. The `top` here may already be `pop`ed out and `heap_.top()` is the second function in the queue, which may be running. Here is a test to reproduce it: https://github.com/jay-zhuang/rocksdb/commit/e46414512a78bf94fdbdde04906c8ff5722a361b

Reply:
Good catch. Will fix.